MEMORY-MANAGEMENT.md

```diff
@@ -0,0 +1,115 @@
+# Memory management in AsyncContext
+
+A context (sometimes called a snapshot; or in the spec, "a List of Async Context
+Mapping Records") is an immutable map from `AsyncContext.Variable` instances to
+arbitrary JS values (or possibly also spec-internal values; see "Using
+AsyncContext from web specs" in the main web integration document). Each agent
+will have a `[[AsyncContextMapping]]` field which is a context, which we will
+sometimes refer to as "the current context".
+
+Given a variable `asyncVar`, which is an instance of `AsyncContext.Variable`,
+running `asyncVar.run(value, callback)` will:
+1. Create a new context which is a copy of the current context, except that
+   `asyncVar` maps to `value`. The reference to `value` in the context is
+   strongly held (not a weak reference), but see the
+   [weak map section](#weak-maps) below.
+2. Set that new context as the current context.
+3. Run the callback.
+4. Restore the current context to the value it had before step 2.
+
+By itself, this would only allow keeping memory alive implicitly within a call
+stack, which would be no different from local variables from a stack frame being
+kept alive while a function is running.
+
+However, the thing that makes AsyncContext AsyncContext is that the context can
+be propagated across asynchronous operations, which eventually cause tasks or
+microtasks to be enqueued. Some of these operations are defined in TC39, such as
+`Promise.prototype.then` and `await`, but most of them are defined in web specs,
+such as `setTimeout` and the many other APIs listed above.
+
+For many of these async operations (such as `setTimeout` and `.then`), a
+callback is run once or multiple times in a task or microtask. In those cases,
+the operation can be seen as keeping a strong reference to the callback, and it
+will also keep a strong reference to the context that was current at the time
+that the API was called to start that operation. When the operation is finished,
+that reference will be removed.
+
+For events, we do not store the context in which `addEventListener` is
+called (though see the next paragraph on the fallback context). Instead, the
+context is propagated from the web API that caused it, if any. For APIs that
+cause events to fire asynchronously (e.g. XHR), this would involve storing a
+reference to the context when the API is called (e.g. `xhr.send()`), and keeping
+it alive until no events can be fired anymore from that asynchronous operation
+(e.g. until the XHR request finishes, errors out or is aborted).
+
+`addEventListener`, however, would need changes with the fallback context
```

Comment:
What do you think about describing the latest iteration of the proposal, and just keep a footnote saying "the old version would capture the whole snapshot, but we are not doing that anymore"?

Reply:
While doing this, I noticed that some of your previous suggestions weren't applied, even though Github says they were. I also made those changes.